protocols/v2/subprotocols/mining/src/new_mining_job.rs

```diff
@@ -51,59 +64,70 @@ impl<'d> NewMiningJob<'d> {
     }
 }
 
-/// NewExtendedMiningJob (Server -> Client)
+/// Message used by an upstream to provide an updated mining job to the downstream through
+/// Extended or Group Channel only.
+///
+/// An Extended Job allows rolling Merkle Roots, giving extensive control over the search space
+/// so that they can implement various advanced use cases such as: translation between Sv1 and Sv2
+/// protocols, difficulty aggregation and search space splitting.
+///
+/// In the Extended Channel scenario, the full coinbase transaction can be constructed as of the
+/// following: `extranonce_prefix + extranonce`, which should equal the negotiated extranonce space
+/// for the channel [`extranonce_size`].
+///
+/// In the Group Channel scenario, the full coinbase can then be constructed as the following:
+/// [`NewExtendedMiningJob::coinbase_tx_prefix`] + `extranonce_prefix` + `extranonce` +
+/// [`NewExtendedMiningJob::coinbase_tx_suffix`].
```

Comment:
the original phrasing was confusing, I don't see why we're providing different descriptions for Extended vs Group Channel.

it also feels wrong to omit that `NewExtendedMiningJob::coinbase_tx_prefix` and `NewExtendedMiningJob::coinbase_tx_suffix` for coinbase reconstruction on Extended Channels

```suggestion
/// The full coinbase can then be constructed as the following:
/// [`NewExtendedMiningJob::coinbase_tx_prefix`] + `extranonce_prefix` + `extranonce` +
/// [`NewExtendedMiningJob::coinbase_tx_suffix`].
```